### PR TITLE
Update documentation.

### PR DESCRIPTION
**Pull request by jaguarfi**
# Description

Additional information on how to run Rubyric locally has been added. In
addition, a brief troubleshooting section has been added.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,64 +4,170 @@ Rubyric is an online tool in which you can create courses and assignments,
...
+
+If you followed the guides provided in the above link. You will find yourself 
+adding some scripts  to the `~/.bashrc` file. However, you can use your favourite
+shell.
+
+You can always verify your default shell by typing the following command in your 
+terminal
+
+```sh 
+ls -l /proc/$$/exe
```
Comment (raphendyr) [inline README.md:20]: `echo $SHELL` should typically be enough, but that works too on Linuxes (assuming procfs is mounted).

File: README.md [hunk 2]
```diff
@@ -4,64 +4,170 @@ Rubyric is an online tool in which you can create courses and assignments,
...
+console enter the following command, which will assist you in assigning the 
+password to the "rails" user.
 
-### Start server
 ```sh
-bin/delayed_job start
-rails server
+postgres=# \password rails
 ```
+Enter the desired password (in this case, your password should be "rails") at
```
Comment (raphendyr) [inline README.md]: I would recommend using ident authentication with database servers (postgresql and mariadb) for better security. Hence the `createdb -O $USER rubyric`

File: README.md (hunk 1 above)
Comment (jaguarfi) [reply README.md:20]: I will add that option too.

File: README.md (hunk 2 above)
Comment (jaguarfi) [reply README.md]: I am using this user for development purposes [database.yml.base](https://github.com/Aalto-LeTech/rubyric/blob/45168fb2fa096adb4a2d7d7edbf0c947488f473c/config/database.yml.base#L3). Nevertheless, I will add one more line below to mention about this "ident authentication"

**Comment by jaguarfi** [comment]
The changes are made and ready to be reviewed. :slightly_smiling_face:

File: README.md (hunk 2 above)
Comment (raphendyr) [reply README.md]: The point is, as long as the database is in the same machine, there is no point to use password. However, if the database is in another machine (e.g. container), then ident authentication is not possible.

Comment (jaguarfi) [reply README.md]: Yes, maybe there is no point in using a password, but I am documenting what exists. In the [current implementation](https://github.com/Aalto-LeTech/rubyric/blob/45168fb2fa096adb4a2d7d7edbf0c947488f473c/config/database.yml.base#L2), they set the username and password. Thus my intention with this PR is not to modify the code but to extend the current documentation, to help the user run Rubyric locally.

I will rephrase the last paragraph I added, and I think that will be enough for documenting the current source code.

File: doc/rubyric.md [hunk 3]
```diff
@@ -84,21 +84,21 @@ Now you can access Rubyric at http://localhost:3000/.
...
 instructions at https://apluslms.github.io/guides/quick/.
 
-In order to get A+ and Rubyrci to communicate with each other we need to do 
+In order to get A+ and Rubyric to communicate with each other we need to do 
 some configuring to both services. At the time of writing this guide A+ needs 
 ports 8000, 8080 and 3000. Thus we cannot use the default port 3000 with 
 Rubyric. Choose some other port for Rubyric, e.g. 3030. You also 
-need to find out your docker ip address, which can be for example something like 
-172.17.0.1 or 172.18.0.1 as A+ should contact that address in order to reach
+need to find out your docker Gateway address, which can be for example something
```
Comment (markkuriekkinen) [inline doc/rubyric.md]: Is "gateway address" a real concept? And why is it capitalized? Anyway, this could be explained in simpler terms. I suppose the main point is that the A+ container has its own IP address and the host machine has its own as well. The IP addresses are in the private address range starting with 172 since they are created by Docker and they do not need to be visible outside the host machine. Can you clarify which address refers to the host machine and which to the A+ container? For example, does the LTI service configuration in A+ refer to the host machine IP address? Can you also mention how one could determine the addresses (maybe with `ifconfig` in the command line)?

File: README.md [hunk 4]
```diff
@@ -4,64 +4,183 @@ Rubyric is an online tool in which you can create courses and assignments,
...
-or
+
+or 
+
 ```sh
 sudo gem install rails -v 5.0.2
 ```
-Rubyric uses postgresql as database. Install postgresql and create user.
+
+It is important to check the Bundler version before start *installing any Gem*.
```
Comment (markkuriekkinen) [inline README.md]: before YOU start

File: README.md [hunk 5]
```diff
@@ -4,64 +4,183 @@ Rubyric is an online tool in which you can create courses and assignments,
...
+```sh
+postgres=# \q
+```
+
+It is important to know that we are using the "rails" user only for developing
+purposes. In a production environment it is better to other means to improve 
+security during the [database authentication](https://www.postgresql.org/docs/9.6/auth-methods.html). 
+
+**Side note:** The rails user can be also modified in the [database.yaml.base](config/database.yml.base) 
+file, and posteriory in the [database.yaml file](#local-configuration) if you wish.
```
Comment (markkuriekkinen) [inline README.md]: `posteriory` is not a real word. "Afterwards" could be used here.

File: README.md [hunk 6]
```diff
@@ -4,64 +4,183 @@ Rubyric is an online tool in which you can create courses and assignments,
...
+
+Now you may need to exit the PostgreSQL console by entering the `\q` command and 
+pressing the enter key.
+
+```sh
+postgres=# \q
+```
+
+It is important to know that we are using the "rails" user only for developing
+purposes. In a production environment it is better to other means to improve
```
Comment (markkuriekkinen) [inline README.md]: to USE other means

File: README.md [hunk 7]
```diff
@@ -70,3 +189,136 @@ or you can create a new user as admin
 ```sh
 User.create(email: email, password: password, admin: '1')
 ```
+
+## Run Rybyric
```
Comment (markkuriekkinen) [inline README.md]: "Rybyric" -> "Rubyric"

File: README.md [hunk 8]
```diff
@@ -70,3 +189,136 @@ or you can create a new user as admin
...
+the following command.
+
+```sh
+bin/delayed_job start
+rails server 
+```
+
+Once the server is started, you can now initialise Rubyric at http://localhost:3000.
+
+If you are planning to run Rubyric along with A+. You may prefer to run Rubyric
```
Comment (markkuriekkinen) [inline README.md]: Comma between the sentences: If you are planning to run Rubyric along with A+, you may prefer to run Rubyric...

File: README.md [hunk 9]
```diff
@@ -70,3 +189,136 @@ or you can create a new user as admin
...
+  Did you mean? yaml_tag
+  ```
+  you may run the following command. 
+
+  ```sh
+  bundle update delayed_job
+  ```
+
+  In addition, you could google this compatibility issue, and try to find the 
+  version of the `delayed_job` that solve this  issue.
```
Comment (markkuriekkinen) [inline README.md]: solve -> solves

**Review by markkuriekkinen — CHANGES_REQUESTED** [review]
I have small suggestions for improvement and then you can squash the commits.

File: doc/rubyric.md (hunk 3 above)
Comment (jaguarfi) [reply doc/rubyric.md]: I think the gateway IP address concept is somehow related to this https://docs.docker.com/network/network-tutorial-standalone/. Therefore, we should use the IP address of the gateway to connect the Rubyric instance to the same network that the docker containers are connected to.

File: doc/rubyric.md [hunk 10]
```diff
@@ -84,48 +84,87 @@ Now you can access Rubyric at http://localhost:3000/.
...
+be something like `"Gateway": "172.17.0.1"`, `"Gateway": "172.18.0.1"`, or
+something similar that starts with the `172.` prefix.
+
+> **NOTE:** In this particular case, we are using the name of the network
+  `aplus_default`, but you should use the name that is shown in your terminal.
+
+
+### Initialise Rubyric
+
+Now that yo know the gateway IP address and the port you want to use to run
```
Comment (markkuriekkinen) [inline doc/rubyric.md]: yo -> you

File: README.md [hunk 11]
```diff
@@ -1,67 +1,198 @@
...
+the most suitable [version](https://rubygems.org/gems/bundler/versions) for you.
+
+```sh
+gem install bundler -v 1.17.3
+```
+
+### Database
+
+Rubyric uses a PostgreSQL database. Therefore, you must install PostgreSQL and
+create an user.
```
Comment (markkuriekkinen) [inline README.md]: "an user" -> "a user" is correct since the pronunciation begins with a consonant

**Review by markkuriekkinen — APPROVED** [review]
This looks good. You could fix the two typos that I found and then squash the commits.

File: doc/rubyric.md (hunk 3 above)
Comment (markkuriekkinen) [reply doc/rubyric.md]: Yes, I agree. The gateway IP address should work for connecting from the A+ container to the Rubyric server running in the host.
